### PR TITLE
Add unit test for TaskManagerFactory and MessgaeProcessorSerializer

### PR DESCRIPTION
**Pull request by Nirothipan**
## Automation tests
 - Unit tests 
   > MessageProcessorSerializer - 88%
  > TaskManagerFactory - 69%

### CODE REVIEW

File: modules/core/src/test/java/org/apache/synapse/config/xml/TaskManagerFactoryTest.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        OMElement element = AXIOMUtil.stringToOM(inputXML);
+        TaskManagerFactory.createTaskManager(element, null);
+    }
+
+    /**
+     * test createTaskManager with provider not specified.
+     *
+     * @throws XMLStreamException - XMLStreamException.
+     */
+    @Test(expected = SynapseException.class)
```
Comment (himasha) [inline modules/core/src/test/java/org/apache/synapse/config/xml/TaskManagerFactoryTest.java]: Can we check for a specific assert message as well? If not how are we going to differentiate from testCreateTaskManager3() and  testCreateTaskManager4()  as they both throw the same Exception Type?

File: modules/core/src/test/java/org/apache/synapse/config/xml/MessageProcessorSerializerTest.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    }
+
+    /**
+     * test serializeMessageProcessor with parameters and assert the OMElement is created.
+     */
+    @Test
+    public void testSerializeMessageProcessor3() {
+        MessageProcessor messageProcessor = new ScheduledMessageForwardingProcessor();
+        messageProcessor.setName("testStore");
+        Map<String, Object> parameters = new HashMap<>();
```
Comment (milindaperera) [inline modules/core/src/test/java/org/apache/synapse/config/xml/MessageProcessorSerializerTest.java]: Shall we initialize this parameters variable at the beginning of the function?

File: modules/core/src/test/java/org/apache/synapse/config/xml/TaskManagerFactoryTest.java (hunk 1 above)
Comment (Nirothipan) [reply modules/core/src/test/java/org/apache/synapse/config/xml/TaskManagerFactoryTest.java]: Fixed this. Thank you

File: modules/core/src/test/java/org/apache/synapse/config/xml/MessageProcessorSerializerTest.java (hunk 2 above)
Comment (Nirothipan) [reply modules/core/src/test/java/org/apache/synapse/config/xml/MessageProcessorSerializerTest.java]: Fixed this.

File: modules/core/src/test/java/org/apache/synapse/config/xml/TaskManagerFactoryTest.java [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+        String inputXML = "<taskManager provider=\"org.apache.synapse.startup.quartz.QuartzTaskManager\">"
+                + "<parameter xmlns=\"http://ws.apache.org/ns/synapse\"  name =\"test\"> test </parameter>"
+                + "</taskManager>";
+        OMElement element = AXIOMUtil.stringToOM(inputXML);
+        TaskManager taskManager = TaskManagerFactory.createTaskManager(element, null);
+        Assert.assertTrue("TaskManager is not created.", taskManager instanceof QuartzTaskManager);
+    }
+
+    @Rule
+    public ExpectedException expectedEx = ExpectedException.none();
```
Comment (nuwanw) [inline modules/core/src/test/java/org/apache/synapse/config/xml/TaskManagerFactoryTest.java]: can you move the variable initialising to beginning of the class

Comment (Nirothipan) [reply modules/core/src/test/java/org/apache/synapse/config/xml/TaskManagerFactoryTest.java]: fixed this

**Pull request merged**